### ISSUE TITLE
Create a test for PointMutationExecutor

### ISSUE THREAD

**Issue by mikemhenry** [choderalab/perses#802]
We don't have a test for `PointMutationExecutor` and a regression occurred when changing the temperature in PR #783. I'm not sure yet why this PR  causes this to fail with `AssertionError: the difference between the atoms_with_positions_reduced_potential and the sum of atoms_with_positions_reduced_potential_components is 15.290058650692572`

``` python
from perses.app.relative_point_mutation_setup import PointMutationExecutor
from simtk import unit
solvent_delivery = PointMutationExecutor("ala_vacuum.pdb",
                        '1',
                        '2',
                        'ASP',
                        ionic_strength=0.15*unit.molar,
                        flatten_torsions=True,
                        flatten_exceptions=True,
                        conduct_endstate_validation=False
                       )
```

But we may have caught this bug if we had a test for this class. 

We should fix the bug (by reverting the PR) and also add the test, but make sure the test fails before reverting the bug when writing the test so we can be sure that the test is testing what we think it is testing.

**Comment by dominicrufa** [comment]
>  I'm not sure yet why this PR causes this to fail with AssertionError: the difference between the atoms_with_positions_reduced_potential and the sum of atoms_with_positions_reduced_potential_components is 15.290058650692572

the explanation is rather involved, but it has to do with `flatten_torsions=True`. there is a special argument in the `geometry.py` that has to be passed to avoid this error. @zhang-ivy has this working locally and probably has a `.py` that can be adapted to a test

**Comment by zhang-ivy** [comment]
@dominicrufa I'm the one that brought this error to Mike's attention. Note that in his original message, `conduct_endstate_validation=False`. The error likely arises because I changed the global temperature variable in `relative_point_mutation_setup.py` to 298 K, but it's still 300 K in geometry.py

**Comment by mikemhenry** [comment]
I used git bisect to find the offending commit and it is indeed when we changed the temperature from 300 K to 289 K. We need to trace through the code and see where temperature comes into our validation code. A quick grep of `unit.kelvin` shows a lot of places where we set temperatures.

**Comment by zhang-ivy** [comment]
@mikemhenry 
Here is the problematic line: https://github.com/choderalab/perses/blob/ac5cd56e138383b557ecfd5d0fc780bef193ec0d/perses/tests/utils.py#L26
Temperature is hard coded as 300 K here, so when we set temperature to 298 K in `relative_point_mutation_setup.py`, the thermodynamic states are inconsistent.

**Comment by mikemhenry** [comment]
Awesome! Thanks for investing this, it will help me with my temperature PR.

**Comment by jchodera** [comment]
This was fixed by #804
